src/Tools-CodeNavigation-Tests/CNSelectorExtractorTest.class.st

```diff
@@ -0,0 +1,97 @@
+"
+A CNSelectorExtractorTest is a test class for testing the behavior of CNSelectorExtractor
+"
+Class {
+	#name : #CNSelectorExtractorTest,
+	#superclass : #TestCase,
+	#classVars : [
+		'selectorExtractor'
+	],
+	#category : #'Tools-CodeNavigation-Tests'
+}
+
+{ #category : #initialization }
+CNSelectorExtractorTest >> initialize [
+	"comment stating purpose of instance-side method"
+	"scope: class-variables  &  instance-variables"	
+			
+	selectorExtractor := CNSelectorExtractor new.
+]
+
+{ #category : #test }
+CNSelectorExtractorTest >> testExtractSelectorFromSourceAtPosition_BorderCases_ReturnsNil [
+
+   |source position|
+	"Border cases first"
+	
+	"Nil source, returns nil"
+	source := nil.
+	position := 1.
+	self assert: (selectorExtractor extractSelectorFromSource: source atPosition: position) isNil.
+	
+	"Empty source, returns nil"
+	source := ''.
+	position := 1.
+	self assert: (selectorExtractor extractSelectorFromSource: source atPosition: position) isNil.
+	
+	"nil source, invalid Position"
+	source := nil.
+	position := -1.
+	self assert: (selectorExtractor extractSelectorFromSource: source atPosition: position) isNil.
+	
+	"a string source, invalid Position"
+	source := 'astring'.
+	position := 10000.
+	self assert: (selectorExtractor extractSelectorFromSource: source atPosition: position) isNil.
+
+]
+
+{ #category : #test }
+CNSelectorExtractorTest >> testExtractSelectorFromSourceAtPosition_GeneralCases_ExtractsSelector [
```

Comment:
I find the underscores in method naming suspicious. 
Usually that convention is not used, and we prefer camel case naming convention for methods.
Could the name of the test method be shorntened to avoid the use of underscores?

Reply:
+1

Reply:
I removed the underscores because it seems like it is not used anywhere else, as Stéphane mentioned.
However, in my opinion, this highlights an issue (A **Naming Convention Issue** in general). Is there an official convention? There are always a reasoning behind convention rules, and we could avoid errors or suggesting naming based on personal preferences (like I did).

The "Tests framework" (Don't know the name) needs a particular name for tests methods to make the "connection" to the actual object method implementation.
I don't like the underscore also, I just used as a reflex. The idea(in previous work) was to have Unit tests with names that clearly describe the purpose by using the tested "methodName_inputOrCase_expectedOutcome" format. I am not saying that this is good or bad for Pharo, I am just saying that the current state of tests is not very good in my opinion.

Having "shortened" names will surely break the current "connection" of the method being tested and its test method. Also, I am not a fan of suggestions from personal preferences.